R/update_hub_target_data.R

```diff
@@ -0,0 +1,119 @@
+nhsn_col_names <- list(
+  covid = "totalconfc19newadm",
+  rsv = "totalconfrsvnewadm"
+)
+nssp_col_names <- list(
+  covid = "percent_visits_covid",
+  rsv = "percent_visits_rsv"
+)
+
+#' Get and process target data for the hub
+#'
+#' This function pulls, formats and save NHSN and NSSP data for use in the hub.
+#'
+#' @param base_hub_path Path to the base hub directory.
+#' @param disease Disease name ("covid" or "rsv").
+#' @param as_of Date of the data pull. Default is today's date.
+#' @param excluded_locations Vector of location codes to exclude from the output.
+#' Default value is c("78", "74", "69", "66", "60"), corresponding to "US Virgin Islands",
+#' "U.S. Minor Outlying Islands", "Guam", "American Samoa", and "Northern Mariana Islands".
+#' @param nhsn_first_weekending_date First week-ending date to include for
+#' the NHSN dataset. Default value is "2024-11-09".
+#' @param legacy_file Logical. Whether to write legacy CSV output (default: FALSE).
+#'
+#' @return Writes `time-series.parquet` and optionally legacy csv target data files
+#' to the target-data directory in the hub.
+#' @export
+update_hub_target_data <- function(
+  base_hub_path,
+  disease,
+  as_of = lubridate::today(),
+  nhsn_first_weekending_date = lubridate::as_date("2024-11-09"),
+  excluded_locations = c("78", "74", "69", "66", "60"),
+  legacy_file = FALSE
+) {
+  if (!disease %in% c("covid", "rsv")) {
+    stop("'disease' must be either 'covid' or 'rsv'")
+  }
+  output_dirpath <- fs::path(base_hub_path, "target-data")
+  fs::dir_create(output_dirpath)
+
+  nhsn_col_name <- nhsn_col_names[[disease]]
+  nssp_col_name <- nssp_col_names[[disease]]
+  hubverse_ts_req_cols <- c(
+    "date",
+    "observation",
+    "location",
+    "as_of",
+    "target"
+  )
+  nhsn_data <- forecasttools::pull_data_cdc_gov_dataset(
+    dataset = "nhsn_hrd_prelim",
+    columns = nhsn_col_name,
+    start_date = nhsn_first_weekending_date
+  ) |>
+    dplyr::mutate(
+      date = lubridate::as_date(.data$weekendingdate),
+      observation = as.numeric(.data[[nhsn_col_name]]),
+      jurisdiction = stringr::str_replace(.data$jurisdiction, "USA", "US"),
+      location = forecasttools::us_location_recode(
+        .data$jurisdiction,
+        "abbr",
+        "code"
+      ),
+      as_of = !!as_of,
+      target = glue::glue("wk inc {disease} hosp")
+    ) |>
+    dplyr::filter(!(.data$location %in% !!excluded_locations))
+
+  hubverse_format_nhsn_data <- nhsn_data |>
+    dplyr::select(dplyr::all_of(hubverse_ts_req_cols))
```

Comment:
nitpick: The `all_of` function technically lives within `tidyselect`, though it's exposed via `dplyr` as well
```suggestion
    dplyr::select(tidyselect::all_of(hubverse_ts_req_cols))
```